google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -44,27 +62,63 @@
 
 /** Implementation of Cloud Spanner remote calls using Gapic libraries. */
 public class GapicSpannerRpc implements SpannerRpc {
+  
   private final SpannerStub stub;
+  private final InstanceAdminStub instanceAdminStub;
+  private final DatabaseAdminStub databaseAdminStub;
+  private final String projectName;
 
   public GapicSpannerRpc() throws IOException {
     stub = GrpcSpannerStub.create(SpannerStubSettings.newBuilder().build());
+    instanceAdminStub = GrpcInstanceAdminStub.create(
+        InstanceAdminStubSettings.newBuilder().build());
+    databaseAdminStub = GrpcDatabaseAdminStub.create(
+        DatabaseAdminStubSettings.newBuilder().build());
+    projectName = "";
   }
 
   @Override
   public Paginated<InstanceConfig> listInstanceConfigs(int pageSize, @Nullable String pageToken)
       throws SpannerException {
-    throw new UnsupportedOperationException("Not implemented yet.");
+    ListInstanceConfigsRequest.Builder requestBuilder = 
+        ListInstanceConfigsRequest.newBuilder().setParent(projectName).setPageSize(pageSize);
+    if (pageToken != null) {
+      requestBuilder.setPageToken(pageToken);
+    }
+    ListInstanceConfigsRequest request = requestBuilder.build();
+
+    // TODO: put projectName in metadata
+    ListInstanceConfigsResponse response = 
+        instanceAdminStub.listInstanceConfigsCallable().call(request);
```

Comment:
Instead of `call(request)`, I think we should use `get(instanceAdminStub.listInstanceConfigsCallable().futureCall(request)`.

`get` was implemented in #3039 and translate some exceptions into `SpannerException`.